### ISSUE TITLE
Support new-style generics on Python 3.9

### ISSUE THREAD

**Issue by tmke8** [wearepal/neoconfigen#31]
In Python 3.9, you can write `list[int]` instead of `List[int]`, but neoconfigen gets confused by that and then produces files like this:

```python
from builtins import List
from dataclasses import dataclass, field
from typing import Optional

@dataclass
class TrainConf:
    ids: Optional[List[int]] = None
```
The problem is that the import `from builtins import List` fails at runtime. It should be `from builtins import list` (though this is unnecessary of course, the builtins are available without an import). I don't know where the capitalization comes from.

**Comment by tmke8** [comment]
This code seems to be the culprit:

https://github.com/wearepal/neoconfigen/blob/0b0653ebee6c0e79a33d226f9a074daed4ecbb73/configen/utils.py#L84-L96

**Comment by tmke8** [comment]
closed by #33
